torchgeo/datasets/south_america_soybean.py

```diff
@@ -0,0 +1,227 @@
+import glob
+import os
+from collections.abc import Iterable
+from typing import Any, Callable, Optional, Union
+
+import torch
+import matplotlib.pyplot as plt
+from matplotlib.figure import Figure
+from rasterio.crs import CRS
+
+from .geo import RasterDataset
+from .utils import BoundingBox, download_url
+
+
+class south_america_soybean(RasterDataset):
+    """South America Soybean Dataset
+
+    Link: https://www.nature.com/articles/s41893-021-00729-z
+
+    Dataset contains 1 classes:
+    1: soybean
+
+    Dataset Format:
+    1) 21 .tif files 
+
+    If you use this dataset in your research, please use the corresponding citation:
+    Song, XP., Hansen, M.C., Potapov, P. et al. Massive soybean expansion in South America since 2000 and implications for conservation. Nat Sustain 4, 784–792 (2021). https://doi.org/10.1038/s41893-021-00729-z 
+
+    """
+    filename_glob = "SouthAmerica_Soybean_*.tif"
+    filename_regex = (r"SouthAmerica_Soybean_(?P<year>\d{4})\.tif")
+
+    zipfile_glob = ""
+
+    date_format = "%Y"
+    is_image = False
+
+    url = "https://glad.umd.edu/projects/AnnualClassMapsV1/SouthAmerica_Soybean_2001.tif"
```

Comment:
```suggestion
    url = "https://glad.umd.edu/projects/AnnualClassMapsV1/SouthAmerica_Soybean_"
```
This url format can fit the for loop for 2001-2021.